Cutelyst/Plugins/Utils/Validator/validatorsame.cpp

```diff
@@ -35,32 +35,30 @@ ValidatorSame::~ValidatorSame()
 {
 }
 
-bool ValidatorSame::validate()
+QString ValidatorSame::validate() const
 {
-    Q_D(ValidatorSame);
+    Q_D(const ValidatorSame);
 
     QString v = value();
 
     if (v.isEmpty()) {
-        setError(ValidatorRule::NoError);
-        return true;
+        return QString();
     }
 
     if (v == d->parameters.value(d->otherField)) {
-        setError(ValidatorRule::NoError);
-        return true;
+        return QString();
     }
 
-    return false;
+    return validationError();
 }
 
-QString ValidatorSame::genericErrorMessage() const
+QString ValidatorSame::genericValidationError() const
 {
     Q_D(const ValidatorSame);
 
     QString ol = !d->otherLabel.isEmpty() ? d->otherLabel : d->otherField;
 
-    return QStringLiteral("The “%1” field must have the same value as the “%2” field.").arg(genericFieldName(), ol);
+    return QStringLiteral("The “%1” field must have the same value as the “%2” field.").arg(fieldLabel(), ol);
```

Comment:
Ok, not sure I was clear and not sure you understand the issue, so let me list the issues:
1 - The CI was failing due two issues, first the unit test file CPP had non latin1 chars, the second was that the validator.dll wasn't next to the test executable.
2 - The issue I'm listing here doesn't fail the CI, because you don't test your error messages, if you do so you will have a failing CI.

On MSVC every file is by default latin1, if it has BOM the compiler understand that it is UTF-8, so a
QStringLiteral(" “ ") without a BOM get's corrupted and the user would see an error message incorrectly.

I could not find a documentation on this (though I remember there is one), the issue is seem on this bugreport https://bugreports.qt.io/browse/QTBUG-52188